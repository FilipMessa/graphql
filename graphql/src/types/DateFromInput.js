// @flow

import { GraphQLInputObjectType, GraphQLBoolean } from 'graphql';
import { GraphQLDate } from 'graphql-iso-date';
import GraphQLDateRange from './DateRangeInput';

export default new GraphQLInputObjectType({
  name: 'DateFromInput',
  fields: {
    exact: {
      type: GraphQLDate,
    },
    range: {
      type: GraphQLDateRange,
    },
    anytime: {
      type: GraphQLBoolean,
    },
  },
});
