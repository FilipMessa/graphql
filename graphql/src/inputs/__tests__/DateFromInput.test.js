// @flow

import DateFromInput from '../DateFromInput';

it('DateFromInput have valid fields', () => {
  expect(DateFromInput.getFields()).toMatchSnapshot();
});
