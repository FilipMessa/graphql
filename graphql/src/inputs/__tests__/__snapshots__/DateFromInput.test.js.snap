// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateFromInput have valid fields 1`] = `
Object {
  "anytime": Object {
    "name": "anytime",
    "type": "Boolean",
  },
  "exact": Object {
    "name": "exact",
    "type": "Date",
  },
  "range": Object {
    "name": "range",
    "type": "DateRangeInput",
  },
}
`;
